force-app/main/default/classes/SOQL.cls

```diff
@@ -342,17 +410,20 @@ public with sharing class SOQL {
 	 */
 	public static void setPrimaryContact() {
 		//Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
+		List<Opportunity> opportunityDetailsForClosedWon = getOpportunityDetailsForClosedWon();
+		// Loop through each opportunity
+		for(Opportunity opp : opportunityDetailsForClosedWon) {
+			if (opp.AccountId != null){
+			// Query the Contacts
+			Contact firstContact = [SELECT FirstName, Id FROM Contact WHERE AccountId = :opp.AccountId ORDER BY FirstName Asc LIMIT 1];
+			if (firstContact != null){
+				opp.Primary_Contact__c = firstContact.Id;
+			}
+		}
+	}
+		update opportunityDetailsForClosedWon;
 
-		// Create a set of Account Ids from the Opportunities
-
-		// Query the Contacts
-
-		// Loop through each Contact and add the Account Id and Contact to the map
-
-		// Loop through each Opportunity
-
-		// Update the Opportunity primary contact from accountToContactMap
```

Comment:
Try to follow these directions step by step in order to remove the soql query from the for loop on line 418. I know there's a limit of 1 but try to avoid them when you can.